app/code/Magento/Catalog/Model/Product/Option.php

```diff
@@ -135,14 +148,34 @@ public function __construct(
         \Magento\Framework\Model\ResourceModel\AbstractResource $resource = null,
         \Magento\Framework\Data\Collection\AbstractDb $resourceCollection = null,
         array $data = [],
-        ProductCustomOptionValuesInterfaceFactory $customOptionValuesFactory = null
+        ProductCustomOptionValuesInterfaceFactory $customOptionValuesFactory = null,
+        array $optionGroups = [],
+        array $optionGroupsToTypes = []
     ) {
         $this->productOptionValue = $productOptionValue;
         $this->optionTypeFactory = $optionFactory;
-        $this->validatorPool = $validatorPool;
         $this->string = $string;
+        $this->validatorPool = $validatorPool;
         $this->customOptionValuesFactory = $customOptionValuesFactory ?:
             \Magento\Framework\App\ObjectManager::getInstance()->get(ProductCustomOptionValuesInterfaceFactory::class);
+        $this->optionGroups = $optionGroups ?: [
+            'date' => 'Magento\Catalog\Model\Product\Option\Type\Date',
```

Comment:
1. Please use `::class` for class names
2. I guess it’s possible to use constants there. Could you use them, or am I wrong?

Reply:
Added constants here and inside di.xml too